packages/vue/src/components/setup-totp.vue

```diff
@@ -18,15 +18,14 @@ const logger = new Logger('SetupTotp-logger');
 
 // `facade` is manually typed to `UseAuthenticator` for temporary type safety.
 const facade: UseAuthenticator = useAuthenticator();
-const { updateForm, submitForm, toSignIn } = facade;
-const { error, isPending, QRFields, totpSecretCode, username } = toRefs(facade);
+const { updateForm, submitForm, toSignIn, totpSecretCode, username, QRFields } =
+  facade;
+const { error, isPending } = toRefs(facade);
 
-const { totpIssuer = 'AWSCognito', totpUsername = username.value } =
-  QRFields.value ?? {};
+const { totpIssuer = 'AWSCognito', totpUsername = username } = QRFields ?? {};
 
-const totpCodeURL = totpSecretCode.value
-  ? getTotpCodeURL(totpIssuer, totpUsername!, totpSecretCode.value)
-  : null;
+// `totpSecretCode` is typed as `string | null` but will always be populated by the machine here
+const totpCodeURL = getTotpCodeURL(totpIssuer, totpUsername, totpSecretCode!);
```

Comment:
Truthiness check of `totpSecretCode.value` is no longer needed. Before migrating to `aws-amplify@6.x` the `totpSecretCode` value needed to be generated by an additional async call, this check was a remnant of the previously required handling.

`totpSecretCode` is now provided directly via the state machine from the payload of a successful call to `signIn`, `signUp`, or `confirmSignUp` depending on the end user's current account status